x/rewards/keeper/iprpc.go

```diff
@@ -178,9 +177,11 @@ func (k Keeper) distributeIprpcRewards(ctx sdk.Context, iprpcReward types.IprpcR
 			UsedReward = UsedRewardTemp
 
 			// reward the provider
-			_, _, err := k.dualstakingKeeper.RewardProvidersAndDelegators(ctx, providerCU.Provider, specFund.Spec, providerIprpcReward, string(types.IprpcPoolName), false, false, false)
+			_, claimableRewards, err := k.dualstakingKeeper.RewardProvidersAndDelegators(ctx, providerCU.Provider, specFund.Spec, providerIprpcReward, string(types.IprpcPoolName), false, false, false)
 			if err != nil {
+				// failed sending the rewards, add the claimable rewards to the leftovers that will be transferred to the community pool
 				utils.LavaFormatError("failed to send iprpc rewards to provider", err, utils.LogAttr("provider", providerCU))
+				leftovers = leftovers.Add(claimableRewards...)
```

Comment:
let's add a unitest for this

Reply:
if this is unreachable code let's panic instead

Reply:
done. see https://github.com/lavanet/lava/commit/0477945caf6cb9cfea6b0c61d8f6caf67141083a